interfaces/api/pipeline/IRelocalizationTrackablePipeline.h

```diff
@@ -0,0 +1,71 @@
+/**
+ * @copyright Copyright (c) 2024 B-com http://www.b-com.com/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SOLAR_RELOCALIZATIONTRACKABLEPIPELINE_H
+#define SOLAR_RELOCALIZATIONTRACKABLEPIPELINE_H
+
+
+#include "api/pipeline/IRelocalizationPipeline.h"
+#include <xpcf/core/helpers.h>
+
+namespace SolAR {
+using namespace datastructure;
+namespace api {
+namespace pipeline {
+
+/**
+ * @class IRelocalizationTrackablePipeline
+ * @brief <B>Defines a relocalization pipeline based on trackable objects.</B>
+ * <TT>UUID: c80e2716-e78d-42c3-ae94-a57ac517bc50</TT>
+ *
+ * This class provides the interface to define a relocalization processing pipeline based on trackable objects.
+ */
+
+class XPCF_CLIENTUUID("76636689-a511-4fee-8c51-bf951147d513") XPCF_SERVERUUID("7dbb07bb-afe0-4ed0-add8-653d46243466")
+    XPCF_GRPC_CLIENT_RECV_SIZE("-1") XPCF_GRPC_CLIENT_SEND_SIZE("-1")
+    IRelocalizationTrackablePipeline : virtual public IRelocalizationPipeline {
```

Comment:
Isn't `IRelocalizationTrackablePipeline` a generalization of `IRelocalizationPipeline`, and is intended to eventually replace it?

If so, maybe it should be a different class, because when it replaces it everywhere,  `IRelocalizationPipeline` will be useless? Or is this temporary and the deletion/merge will happen later?